hook-worker/src/worker.rs

```diff
@@ -379,26 +389,35 @@ async fn send_webhook(
         .body(body)
         .send()
         .await
-        .map_err(|e| WebhookError::RetryableRequestError {
+        .map_err(|e| WebhookRequestError::RetryableRequestError {
             error: e,
+            response: None,
             retry_after: None,
         })?;
 
     let retry_after = parse_retry_after_header(response.headers());
 
-    match response.error_for_status() {
-        Ok(response) => Ok(response),
+    match response.error_for_status_ref() {
+        Ok(_) => Ok(response),
```

Comment:
Notice we are using a reference to response here when matching as `error_for_status` would have taken ownership of the response, and we need to keep ownership if we intend to return it.